tsdb/chunks/head_chunks_test.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chunks
+
+import (
+	"encoding/binary"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	"github.com/prometheus/prometheus/util/testutil"
+)
+
+func TestHeadReadWriter_WriteChunk(t *testing.T) {
```

Comment:
Though they might look similar in some parts, they are functionally testing different things in a slightly different way. Combining them is possible, but it would get messy/confusing real quick. I would like to keep them separate for simplicity.

Reply:
I am also a fan of table tests, but if you think there is no way to make them generic - fine to keep it separated :+1: